In [1]:
suppressPackageStartupMessages(library(rstanarm))
suppressPackageStartupMessages(library(ggformula))
library(tibble)
suppressPackageStartupMessages(library(glue))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(modelr))
library(stringr)

In [2]:
# Set the maximum number of columns and rows to display
options(repr.matrix.max.cols=150, repr.matrix.max.rows=200)
# Set the default plot size
options(repr.plot.width=18, repr.plot.height=12)

In [3]:
download_if_missing <- function(filename, url) {
    if (!file.exists(filename)) {
        dir.create(dirname(filename), showWarnings=FALSE, recursive=TRUE)
        download.file(url, destfile = filename, method="curl")
    }
}

# Plotting linear and quadratic regressions

The folder [`Earnings`](https://github.com/avehtari/ROS-Examples/tree/master/Earnings/) has data on weight (in pounds), age (in years), and other information from a sample of American adults.
We create a new variable, `age10` = age/10, and fit the following regression predicting weight:

```
            Median   MAD_SD
(Intercept)  148.7     2.2
age10          1.8     0.5

Auxiliary parameter(s)
      Median   MAX_SD
sigma  34.5       0.6
```

## Sketching the model

With pen on paper, sketch a scatterplot of weights versus age (that is, weight on y-ais, age on x-axis) that is consistent with the above information, also drawing the fitted regression line.
Do this just given the information here and your general knowledge about adult heights and weights: do not download the data.

## Adding a quadratic term

Next, we define `age10_sq` $= (\rm{age}/10)^2$ and predict weight as a quadratic function of age:

```
            Median   MAD_SD
(Intercept)  108.0     5.7
age10         21.3     2.6
age10sq       -2.0     0.3

Auxiliary parameter(s)
      Median   MAX_SD
sigma  33.9       0.6
```

Draw this fitted curve on the graph you already sketched above

# Plotting regression with a continuous variable broken into categories

Continuing Exercise 12.1, we divide age into 4 categories and create corresponding inticator variable, `age18_29`, `age30_44`, `age45_64`, and `age65_up`. We then fit the following regression:

```
stan_glm(weight ~ age30_44 + age45_64 + age65_up, data=earnings)

             Median   MAD_SD
(Intercept)   147.8     1.6
age30_44TRUE    9.6     2.1
age45_64TRUE   16.6     2.3
age65_upTRUE    7.5     2.7

Auxiliary parameter(s)
      Median   MAX_SD
sigma  34.1       0.6
```

## Missing Indicator

Why did we not include an indicator for the youngest group, `age18_29`?

## Sketch the graph

Using the same axes and scale as in your graph for Exercise 12.1, sketch with pen on paper the scatterplot, along with the above regression function, which will be discontinuous.

### Check

In [11]:
filename <- "./data/Earnings/earnings.csv"

download_if_missing(filename,
                    'https://raw.githubusercontent.com/avehtari/ROS-Examples/master/Earnings/data/earnings.csv')
earnings <- read.csv(filename, header=TRUE)

# Scale of regression coefficients

A regression was fit to data from different countries, predicting the rate of civil conflicts given a set of geographic and political predictors.
Here are the estimated coefficients and their z-scores (coefficient divided by standard error), given to three decimal places.

|  | Estimate | z-score |
| --- | --- | --- |
| Intercept | -3.814 | -20.178 |
| Conflict before 2000 | 0.020 | 1.861 |
| Distance to border | 0.000 | 0.450 |
| Distance to capital | 0.000 | 1.629 |
| Population | 0.000 | 2.482 |
| % mountainous | 1.641 | 8.518 |
| % irrigated | -0.027 | -1.663 |
| GDP per capita | -0.000 | -3.589 |

Why are the coefficients for distance to border, distance to capital, population, and GDP per capita so small?

# Coding a predictor as both cateogical and continuous

A linear regression is fit on a group of employed adults, predicting their physical flexibility given age.
Flexibility is defined on a 0-30 scale based on measurements from a series of stretching tasts.
Your model includes age in categories (under 30, 30-44, 45-59, 60+) and also age as a linear predictor.
Sketch a graph of flexibility vs. age, showing what the fitted regression might look like.

## Checking with a simulation

# Logarithmic transformation and regression

Consider the following regression:

$$ \log(\rm{weight}) = - 3.8 + 2.1 \log(\rm{height}) + \rm{error}, $$

with errors that have standard deviation 0.25.
Weights are in pounds and heights are in inches.

## Fill in the blanks

Approximately 68% of people will heave weights within a factor of **exp(-0.25)=0.78** and **exp(0.25)=1.3** of their predicted values from the regression.

## Sketch

Using a pen and paper, sketch the regression line and scaterrplot or log(weight) versus log(height) that make sense and are consistent with the fitted model.
Be sure to label the axes of your graph.

# Logarithmic transformations

The folder [`Pollution`](https://github.com/avehtari/ROS-Examples/tree/master/Pollution/) contains mortality rates and various environmental factors from 60 U.S. metropolitan areas (see [McDonald and Schwing, 1973](https://www.tandfonline.com/doi/abs/10.1080/00401706.1973.10489073)).
For this exercise we shall model mortality rate given nitric oxides, sulfur dioxide, and hydrocarbons as inputs.
This model is an extreme oversimplification, as it combines all sources of mortality and does not adjust for crucial factors such as age and smoking.
We use it to illustrate log transformations in regression.

In [14]:
filename <- "./data/Polution/pollution.csv"

download_if_missing(filename,
                    'https://raw.githubusercontent.com/avehtari/ROS-Examples/master/Pollution/data/pollution.csv')
pollution <- read.csv(filename, header=TRUE)

In [17]:
pollution

prec jant jult ovr65 popn educ hous dens nonw wwdrk poor hc  nox so2 humid
1  36   27   71    8.1  3.34 11.4 81.5 3243  8.8 42.6  11.7  21  15  59 59   
2  35   23   72   11.1  3.14 11.0 78.8 4281  3.5 50.7  14.4   8  10  39 57   
3  44   29   74   10.4  3.21  9.8 81.6 4260  0.8 39.4  12.4   6   6  33 54   
4  47   45   79    6.5  3.41 11.1 77.5 3125 27.1 50.2  20.6  18   8  24 56   
5  43   35   77    7.6  3.44  9.6 84.6 6441 24.4 43.7  14.3  43  38 206 55   
6  53   45   80    7.7  3.45 10.2 66.8 3325 38.5 43.1  25.5  30  32  72 54   
7  43   30   74   10.9  3.23 12.1 83.9 4679  3.5 49.2  11.3  21  32  62 56   
8  45   30   73    9.3  3.29 10.6 86.0 2140  5.3 40.4  10.5   6   4   4 56   
9  36   24   70    9.0  3.31 10.5 83.2 6582  8.1 42.5  12.6  18  12  37 61   
10 36   27   72    9.5  3.36 10.7 79.3 4213  6.7 41.0  13.2  12   7  20 59   
11 52   42   79    7.7  3.39  9.6 69.2 2302 22.2 41.3  24.2  18   8  27 56   
12 33   26   76    8.6  3.20 10.9 83.4 6122 16.3 44.9  10.7  88  63 278 58   
13 40   34   77    9.2  3.21 10.2 77.0 4101 13.0 45.7  15.1  26  26 146 57   
14 35   28   71    8.8  3.29 11.1 86.3 3042 14.7 44.6  11.4  31  21  64 60   
15 37   31   75    8.0  3.26 11.9 78.4 4259 13.1 49.6  13.9  23   9  15 58   
16 35   46   85    7.1  3.22 11.8 79.9 1441 14.8 51.2  16.1   1   1   1 54   
17 36   30   75    7.5  3.35 11.4 81.9 4029 12.4 44.0  12.0   6   4  16 58   
18 15   30   73    8.2  3.15 12.2 84.2 4824  4.7 53.1  12.7  17   8  28 38   
19 31   27   74    7.2  3.44 10.8 87.0 4834 15.8 43.5  13.6  52  35 124 59   
20 30   24   72    6.5  3.53 10.8 79.5 3694 13.1 33.8  12.4  11   4  11 61   
21 31   45   85    7.3  3.22 11.4 80.7 1844 11.5 48.1  18.5   1   1   1 53   
22 31   24   72    9.0  3.37 10.9 82.8 3226  5.1 45.2  12.3   5   3  10 61   
23 42   40   77    6.1  3.45 10.4 71.8 2269 22.7 41.4  19.5   8   3   5 53   
24 43   27   72    9.0  3.25 11.5 87.1 2909  7.2 51.6   9.5   7   3  10 56   
25 46   55   84    5.6  3.35 11.4 79.7 2647 21.0 46.9  17.9   6   5   1 59   
26 39   29   75    8.7  3.23 11.4 78.6 4412 15.6 46.6  13.2  13   7  33 60   
27 35   31   81    9.2  3.10 12.0 78.3 3262 12.6 48.6  13.9   7   4   4 55   
28 43   32   74   10.1  3.38  9.5 79.2 3214  2.9 43.7  12.0  11   7  32 54   
29 11   53   68    9.2  2.99 12.1 90.6 4700  7.8 48.9  12.3 648 319 130 47   
30 30   35   71    8.3  3.37  9.9 77.4 4474 13.1 42.6  17.7  38  37 193 57   
31 50   42   82    7.3  3.49 10.4 72.5 3497 36.7 43.3  26.4  15  18  34 59   
32 60   67   82   10.0  2.98 11.5 88.6 4657 13.5 47.3  22.4   3   1   1 60   
33 30   20   69    8.8  3.26 11.1 85.4 2934  5.8 44.0   9.4  33  23 125 64   
34 25   12   73    9.2  3.28 12.1 83.1 2095  2.0 51.9   9.8  20  11  26 58   
35 45   40   80    8.3  3.32 10.1 70.3 2682 21.0 46.1  24.1  17  14  78 56   
36 46   30   72   10.2  3.16 11.3 83.2 3327  8.8 45.3  12.2   4   3   8 58   
37 54   54   81    7.4  3.36  9.7 72.8 3172 31.4 45.5  24.2  20  17   1 62   
38 42   33   77    9.7  3.03 10.7 83.5 7462 11.3 48.7  12.4  41  26 108 58   
39 42   32   76    9.1  3.32 10.5 87.5 6092 17.5 45.3  13.2  29  32 161 54   
40 36   29   72    9.5  3.32 10.6 77.6 3437  8.1 45.5  13.8  45  59 263 56   
41 37   38   67   11.3  2.99 12.0 81.5 3387  3.6 50.3  13.5  56  21  44 73   
42 42   29   72   10.7  3.19 10.1 79.5 3508  2.2 38.8  15.7   6   4  18 56   
43 41   33   77   11.2  3.08  9.6 79.9 4843  2.7 38.6  14.1  11  11  89 54   
44 44   39   78    8.2  3.32 11.0 79.9 3768 28.6 49.5  17.5  12   9  48 53   
45 32   25   72   10.9  3.21 11.1 82.5 4355  5.0 46.4  10.8   7   4  18 60   
46 34   32   79    9.3  3.23  9.7 76.8 5160 17.2 45.1  15.3  31  15  68 57   
47 10   55   70    7.3  3.11 12.1 88.9 3033  5.9 51.0  14.0 144  66  20 61   
48 18   48   63    9.2  2.92 12.2 87.7 4253 13.7 51.2  12.0 311 171  86 71   
49 13   49   68    7.0  3.36 12.2 90.7 2702  3.0 51.9   9.7 105  32   3 71   
50 35   40   64    9.6  3.02 12.2 82.5 3626  5.7 54.3  10.1  20   7  20 72   
51 45   28   74   10.6  3

## Linearity in Nitric Oxides

Create a scatterplot of mortality rate versus level of nitric oxides.
Do you think linear regression will fit these data well?
Fit the regression and evaluate a residual plot from the regression.

## Transforming the data

Find an appropriate transformation that will result in data more appropriate for linear regression.
Fit a regression to the transformed data and evaluate the new residual plot.

## Intepreting Models on Transformed Data

Interpret the slope coefficient from the model you chose in (b).

## Adding Sulfure Dioxide and Hydrocarbons as Predictors

Now fit a model predicting mortality rate using levels of nitric oxides, sulfur dioxide, and hydrocarbons as inputs.
Use appropriate transformations when helpful.
Plot the fitted regression model and interpred the coefficients.

## Cross validate

Fit the model you chose above to the first half of the data and then predict for the second half.
You used all the data to construct the model in (d), so this is not really cross validation, but it gives a sense of how the steps of cross validation can be implemented.

# Cross validation comparison of models with different transformations of outcomes

When we compare models with transformed continuous outcomes, we must take into account how the nonlinear transformation warps the continuous variable.
Follow the procedure used to compare models for the mesquite bushes example on page 202.

## Earnings under log transformation

Compare models for eachings and for log(earnings) given height and sex as shown on pages 84 and 192.
Use `earnk` and `log(earnk)` as outcomes.

In [16]:
earnings

height weight male earn   earnk ethnicity education mother_education
1    74     210    1    50000  50.00 White     16        16              
2    66     125    0    60000  60.00 White     16        16              
3    64     126    0    30000  30.00 White     16        16              
4    65     200    0    25000  25.00 White     17        17              
5    63     110    0    50000  50.00 Other     16        16              
6    68     165    0    62000  62.00 Black     18        18              
7    63     190    0    51000  51.00 White     17        17              
8    64     125    0     9000   9.00 White     15        15              
9    62     200    0    29000  29.00 White     12        12              
10   73     230    1    32000  32.00 White     17        17              
11   72     176    1     2000   2.00 Hispanic  15        15              
12   72     265    1    35000  35.00 White     NA        99              
13   72     160    1    27000  27.00 White     12        12              
14   70     225    1     6530   6.53 White     16        16              
15   63     107    0        0   0.00 White     14        14              
16   68     170    1    30000  30.00 White     11        NA              
17   68     144    1    12000  12.00 White     12        12              
18   71     193    1    15000  15.00 White     18        18              
19   65     250    0    12000  12.00 White     12        12              
20   66     130    0    20000  20.00 White     11        11              
21   60     100    0        0   0.00 White     12        12              
22   65     123    0    25000  25.00 White     12        12              
23   66     130    0    22000  22.00 White     16        16              
24   65     120    0    25000  25.00 White     16        16              
25   71     160    1    20000  20.00 White     14        14              
26   68     172    1    17000  17.00 White     12        12              
27   68     135    0    40000  40.00 White     14        14              
28   70     180    1    44000  44.00 White     13        13              
29   67     112    0        0   0.00 White      9         9              
30   64     240    0     7000   7.00 Black     12        12              
31   73     190    1    53000  53.00 Black     13        13              
32   64     145    0     5000   5.00 Black     12        NA              
33   62     120    0     5000   5.00 White     13        13              
34   63     114    0    14000  14.00 White     14        14              
35   69     170    1    45000  45.00 White     14        14              
36   67     166    1     5500   5.50 White     14        14              
37   66     124    0    40000  40.00 White     12        12              
38   72     175    1    34000  34.00 White     12        12              
39   63     120    0     5000   5.00 Black     12        12              
40   63     184    0    10000  10.00 Black     12        12              
41   68     150    1        0   0.00 White     11        11              
42   64     150    0    27000  27.00 White     16        16              
43   73     220    1    62000  62.00 White     14        14              
44   67     220    0    15000  15.00 White     12        12              
45   60      NA    0        0   0.00 White     12        12              
46   72     180    1    50000  50.00 White     16        16              
47   68     158    0        0   0.00 White     12        NA              
48   77     255    1    41000  41.00 White     16        16              
49   64     150    0        0   0.00 White     10        10              
50   64     130    0    15000  15.00 White     14        14              
51   64      NA    0    25000  25.00 White     12        12              
52   63     146    0    25000  25.00 White     16        16              
53   72     185    1    75000  75.00 White     17        17              
54   68  

## Other examples

Compare models from exercise 12.6.

# Log-log transformations

Suppose that, for a certain population of animals, we can predict log weight from log height as follows:

* An animal that is 50 centimeters tall is predicted to weigh 10 kg.
* Every increase of 1% in height corresponds to a predicted increase of 2% in weight.
* The weights of approximately 95% of the animals fall within a factor of 1.1 of predicted values.

## Description to model

Give the equation of the regression line and the residual standard deviation of the regression.

## Calculating $R^2$

Suppose the standard deviation of log weights is 20% in this population.
What, then, is the $R^2$ of the regression model described here?

# Linear and logarithmic transformations

For a study of congressional elections, you would like a measure of the relative amound of money raised by each of the two major-party candidates in each district.
Suppose that you know the amount of money raised by each candidate; label these doallar values $D_i$ and $R_i$.
You would like to combine these into a single variable that can be included as an input variables into a model predicting vote share for the Democrats.
Discuss the advantages and disadvandates of the following measures:

1. The simple difference, $D_i - R_i$
2. The ratio, $D_i/R_i$
3. The difference on the logarithmic scale, $\log D_i - \log R_i $
4. The relative proportion, $D_i / (D_i + R_i)$.

# Special-purpose transformations

For the congressional elections example in the previous exercise, propose an idiosyncratic transformation as in the example on page 196 and discuss the advantages and disadvantages of using it as a regression input.

# Elasticity
An economist runs a regression examining the relations between the average price of cigarettes, $P$, and the quantity purchages, $Q$, across a large sample of countries in the Unites States, assuming the functional form, $\log Q = \alpha + \beta \log P$.
Suppose the estimate for $\beta$ is 0.3.
Interrpet this coefficient.

# Sequence of regressions

Fina  regression problem that is of interest to you and can be performed repeatedly (for example, data from several years, or for several countries).
Perform a separate analysis for each year, or country, and display the estimates in a plot as in Figure 10.9.

# Building regression models

Return to the teaching evaluations data from Exercise 10.6.
Fit regression models predicting evaluations given many of the inputs in the dataset.
Consider interactions, combinations of predictors, and transformations, as appropriate.
Consider several models, discuss in detail the final model that you choose, and also explain why you chose it rather than the others you had considered.

In [18]:
filename <- "./data/Beauty/beauty.csv"

download_if_missing(filename,
                    'https://raw.githubusercontent.com/avehtari/ROS-Examples/master/Beauty/data/beauty.csv')
beauty <- read.csv(filename)

beauty

eval beauty     female age minority nonenglish lower course_id
1   4.3   0.2015666 1      36  1        0          0      3       
2   4.5  -0.8260813 0      59  0        0          0      0       
3   3.7  -0.6603327 0      51  0        0          0      4       
4   4.3  -0.7663125 1      40  0        0          0      2       
5   4.4   1.4214450 1      31  0        0          0      0       
6   4.2   0.5002196 0      62  0        0          0      0       
7   4.0  -0.2143501 1      33  0        0          0      4       
8   3.4  -0.3465390 1      51  0        0          0      0       
9   4.5   0.0613435 1      33  0        0          0      0       
10  3.9   0.4525679 0      47  0        0          0      4       
11  3.1   0.1432643 0      35  1        1          0      4       
12  4.0  -0.1550228 0      37  0        0          0      0       
13  3.8   0.1285433 0      42  0        0          0      0       
14  3.4  -0.3470453 0      49  0        1          0      0       
15  2.9   0.4619388 1      37  0        0          0      9       
16  4.5  -0.1503849 0      45  0        0          0      9       
17  4.0  -1.0707340 1      56  0        0          0     10       
18  3.8  -0.1426931 0      48  0        0          1      0       
19  4.3  -0.1563634 1      46  0        0          0      0       
20  3.4  -0.0589354 1      57  0        0          0      0       
21  3.4   0.1511368 1      52  0        0          0      9       
22  3.3  -0.9370760 1      29  1        1          0      8       
23  4.3  -0.8214405 0      62  0        0          0     10       
24  4.4  -0.1994712 0      64  0        0          0      0       
25  4.6   1.6871670 1      34  0        0          0      0       
26  4.2  -0.4101976 0      58  0        0          0     25       
27  4.0   0.1233448 0      52  1        1          0     17       
28  3.3  -0.8059942 0      73  0        0          0     13       
29  4.4  -0.8210418 0      70  0        0          0     19       
30  2.3   0.3198185 1      41  0        0          0     15       
31  3.5  -0.0985568 0      63  0        0          0     12       
32  4.7  -1.0123060 0      47  0        0          0      0       
33  3.9   0.4883314 0      39  0        0          1      0       
34  4.0  -0.1719505 1      47  1        0          1     21       
35  4.3  -1.1679070 0      54  0        0          0     16       
36  3.8   0.9525534 1      44  1        0          0      0       
37  3.8  -1.1404570 0      47  1        0          0     16       
38  4.0  -0.8161172 0      62  0        0          0      0       
39  4.6  -0.4837456 0      60  0        0          0      0       
40  3.5   0.8450468 0      37  0        0          0     14       
41  4.7  -0.3261277 0      42  0        0          0      0       
42  4.2   0.1865589 0      35  0        0          0     22       
43  3.4   1.8816740 1      39  0        0          1     21       
44  3.9   0.9626006 0      49  0        0          0     22       
45  4.5   0.1534694 0      61  0        0          1     21       
46  4.7   1.2610460 0      33  0        0          0     22       
47  3.5   0.0232144 1      58  0        0          0     23       
48  3.7  -0.3460746 1      56  0        0          0     22       
49  4.0  -0.7446180 1      50  0        0          1     21       
50  4.1  -0.7447439 0      52  0        0          0      0       
51  4.4   0.6361284 1      33  0        0          0      0       
52  4.2   0.5434852 0      57  0        0          0      0       
53  4.5   0.9825948 1      38  0        0          0      1       
54  3.8  -1.5388430 1      34  0        0          1      0       
55  3.1   1.0675310 0      34  0        0          0      0       
56  3.7  -0.4143639 0      32  0        0          1      0       
57  3.9   0.7962944 1      42  0        0          1      0       
58  4.0  -0.6466943 1      43  0        0          1      0       
59  3.3  -0.4299034 0      35  0        1          

# Prediction from a fitted regression

Consider one of the fitted models for mesquite leaves, for example `fit_4`, in Section 12.6.
Suppose you wish to use this model to make inferences about the average mesquite yield in a new set of trees whose predictors are in data frame called `new_trees`.
Give R code to obtain an estimate and standard error for this population average.

You do not need to make the prediction; just give the code.

In [21]:
?read.table

In [31]:
filename <- "./data/Mesquite/mesquite.dat"

download_if_missing(filename,
                    'https://raw.githubusercontent.com/avehtari/ROS-Examples/master/Mesquite/data/mesquite.dat')
mesquite <- read.table(filename, header=TRUE)

mesquite <- mesquite %>%
mutate(canopy_volume = diam1 * diam2 * canopy_height,
       canopy_area = diam1 * diam2,
       canopy_shape = diam1 / diam2)

In [32]:
fit_4 <- stan_glm(formula=log(weight) ~ log(canopy_volume) + log(canopy_area) + log(canopy_shape) +
                                             log(total_height) + log(density) + group,
                  data=mesquite, refresh=0)

In [34]:
new_trees <- sample(mesquite, 20, TRUE)

# Models for regression coefficients

Using hte Portugese student data from the [`Student`](https://github.com/avehtari/ROS-Examples/tree/master/Student) folder, repeat the analyses in Section 12.7 with the same predictors, but using as outcome the Poruguese language grade rather than the mathematics grade.

In [36]:
filename <- "./data/Student/student-merged.csv"

download_if_missing(filename,
                    'https://raw.githubusercontent.com/avehtari/ROS-Examples/master/Student/data/student-merged.csv')
student <- read.csv(filename)

student

G1mat G2mat G3mat G1por G2por G3por school sex age address famsize Pstatus
1    7    10    10    13    13    13    0      0   15  0       0       1      
2    8     6     5    13    11    11    0      0   15  0       0       1      
3   14    13    13    14    13    12    0      0   15  0       0       1      
4   10     9     8    10    11    10    0      0   15  0       0       1      
5   10    10    10    13    13    13    0      0   15  0       0       1      
6   12    12    11    11    12    12    0      0   15  0       0       1      
7   12     0     0    10    11    12    0      0   15  0       0       1      
8    8     9     8    11    10    11    0      0   15  0       1       1      
9   16    16    16    15    15    15    0      0   15  0       1       1      
10  10    11    11    10    10    10    0      0   15  1       0       0      
11  14    15    15    15    14    15    0      0   15  1       0       0      
12   8     0     0    11    12    13    0      0   15  1       0       1      
13   9    10    10    13    12    12    0      0   15  1       0       1      
14   8    11    11    12    12    12    0      0   15  1       0       1      
15  10    11    11    13    12    13    0      0   15  1       0       1      
16  16    15    15    14    14    14    0      0   15  1       0       1      
17  10    12    12    10    12    13    0      0   15  1       0       1      
18   7     8     8    10    10    10    0      0   15  1       0       1      
19   8     7     6    11    11    11    0      0   15  1       0       1      
20   6     7     0    11    11    11    0      0   15  1       0       1      
21   7     6     6    11    10    10    0      0   15  1       0       1      
22  15    14    15    14    14    14    0      0   15  1       0       1      
23  13    14    14    15    15    15    0      0   15  1       0       1      
24  16    17    18    14    13    13    0      0   15  1       0       1      
25  11     0     0    13    14    15    0      0   15  1       0       1      
26   8    10    11    13    12    12    0      0   15  1       0       1      
27   7     9     8    12    11    12    0      0   15  1       0       1      
28   7     7     7    13    12    12    0      0   15  1       0       1      
29  10     8     9    14    14    14    0      0   15  1       0       1      
30   9    11    11    13    13    13    0      0   15  1       0       1      
31  16    16    15    13    14    14    0      0   15  1       0       1      
32  10    13    13    13    12    13    0      0   15  1       1       0      
33   7    10    11    10    11    11    0      0   15  1       1       0      
34   8     8     6    10    11    11    0      0   15  1       1       0      
35   7     8    10    12    13    12    0      0   15  1       1       1      
36   7     6     6    12    11    11    0      0   15  1       1       1      
37  11    13    13    16    14    16    0      0   15  1       1       1      
38  10    10    10    13    12    12    0      0   15  1       1       1      
39  10    10    11    13    13    13    0      0   16  0       0       1      
40   7    10    10    11    11    11    0      0   16  0       0       1      
41  10    11    11    17    16    16    0      0   16  0       0       1      
42   8     9    10    12    13    12    0      0   16  1       0       0      
43  12    13    14    13    13    13    0      0   16  1       0       0      
44   7     7     7    11    11    12    0      0   16  1       0       0      
45   4     0     0     9     9    10    0      0   16  1       0       0      
46  12    11    11    15    13    14    0      0   16  1       0       0      
47   9     9    10    12    13    13    0      0   16  1       0       1      
48  10     8    11    10    10    16    0      0   16  1       0       1      
49   8     7     0    14    13    13    0      0   16  1       0       1      
50   8     9    10    13    12    12    0      0   16 

# Applying ideas of regression

Read a published article that uses regression modeling and is on a topic of interest to you.
Write a page or two evaluating and criticizing the article, addressing issues discussed in Chapters 1-12, such as measurement, data visualization, modeling, inference, simulation, regression, assumptions, model checking, interactions, and transformations.
The point of this exercise is not to come up with a comprehensive critique of the article but rather to review the key points of this book so far in the context of a live example.

Perhaps [Pierre Chandon and Brian Wansink, “When Are Stockpiled Products Consumed Faster? A Convenience-Salience Framework of Postpurchase Consumption Incidence and Quantity,” *Journal of Marketing Research* 39, no. 3 (August 2002): 321–35.](https://faculty.insead.edu/pierre-chandon/documents/Article-When%20are%20stockpiled%20products%20consumed%20faster%20-%20A%20convenience%20salience%20framework%20of%20post%20purchase%20consumption%20incidence%20and%20quantity.pdf)

Or [Determinants of Store Price Elasticity](https://www.researchgate.net/profile/Peter-Rossi-4/publication/237130017_Determinants_of_Store-Level_Price_Elasticity/links/55e089cc08ae2fac471bea66/Determinants-of-Store-Level-Price-Elasticity.pdf)

[Price Elasticity of SA Electricity (Hyndman)](https://www.monash.edu/business/ebs/our-research/publications/ebs/wp16-10.pdf) hmm..

[Ebook Price Elasticity](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.688.6731&rep=rep1&type=pdf)

[Happiness and Income (Kahneman)](https://www.pnas.org/content/pnas/107/38/16489.full.pdf?source=post_page---------------------------)